src/controllers/hid/hidcontroller.cpp

```diff
@@ -295,6 +239,29 @@ int HidController::close() {
     return 0;
 }
 
+bool HidController::poll() {
+    Trace hidRead("HidReader poll");
+
+    for (int i = 0; i < 10; i++) {
+        int result = hid_read(m_pHidDevice, m_pPollData, sizeof(m_pPollData) / sizeof(m_pPollData[0]));
+        if (result == -1) {
+            return false;
+        } else if (result == 0) {
+            break;
```

Comment:
`return true;`

At the bottom of the function you may then add a debug message that polling has been aborted after the maximum number of consecutive read operations has been reached. Traceability is key for analyzing any issues later! The debug log message doesn't hurt if we don't expect to ever reach this point.